java/memory/memory-core/src/main/java/org/apache/arrow/memory/LowCostIdentityHashMap.java

```diff
@@ -317,7 +317,7 @@ public boolean isEmpty() {
    * @return the number of mappings in this Map.
    */
   public int size() {
-    return size;
```

Comment:
Would it be okay to introduce a `sizeAsLong` method to be introduced and mark `size` function as deprecated.

Reply:
No, we don't want to replace everything. (Also I would expect `longSize`)